frontend/packages/console-dynamic-plugin-sdk/src/api/flag.ts

```diff
@@ -0,0 +1,41 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+
+export const useFlag: UseFlag = (flag) =>
+  useSelector<RootState, boolean>(({ FLAGS }) => FLAGS.get(flag));
+
+type UseFlag = (flag: string) => boolean;
```

Comment:
Let's follow the existing convention of having one file per hook, i.e. `console-dynamic-plugin-sdk/src/api/useFlag.ts`.

The `FLAGS` enum below can be moved into `console-dynamic-plugin-sdk/src/constants.ts`.